### PR TITLE
RPC: payincoinjoin, listpaymentsincoinjoin, cancelpaymentincoinjoin

### PR DESCRIPTION
**Pull request by MarnixCroes**

### CODE REVIEW

File: docs/using-wasabi/RPC.md [hunk 1]
```diff
@@ -630,6 +630,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+  "id": "1"
+}
+```
+
+The first parameter is the destination address, the second parameter is the amount in sats.
+The _result_ is the paymentId.
+
+A _payincoinjoin_ is written to the logs and it's status can be seen by using the _listpaymentsincoinjoin_ method.
+
+Payments in coinjoin can in theory be made to any ScriptPubKey, however the zkSNACKs coordinator currently only accepts P2WPKH and P2TR outputs.
```
Comment (MarnixCroes) [inline docs/using-wasabi/RPC.md:651]: changed this in last push.
previously it was something like _it is only possible for native segwit_
this is better and can later easily add P2PKH P2WSH P2SH etc.

File: docs/using-wasabi/RPC.md [hunk 2]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+
+Currently the default maximum is 4 payments per client per coinjoin.
+
+_payincoinjoin_ only registers a payment, so if coinjoin is not running or the amount is lower than the wallet balance, the payment is queued.
+
+Pending payments can be removed by using the _cancelpaymentincoinjoin_ method.
+Pending payments are also removed if the Wasabi client restarts.
+
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:649]:
```suggestion
```

File: docs/using-wasabi/RPC.md [hunk 3]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
 
 The first parameter is the wallet password, the second parameter is `stopWhenAllMixed`, and the third one is`overridePlebStop`.
 
+### payincoinjoin
+
+Pay to a bitcoin address in a coinjoin.
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:625]:
```suggestion
Pay to a specific bitcoin address in a coinjoin.
```

File: docs/using-wasabi/RPC.md [hunk 4]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+
+_payincoinjoin_ only registers a payment, so if coinjoin is not running or the amount is lower than the wallet balance, the payment is queued.
+
+Pending payments can be removed by using the _cancelpaymentincoinjoin_ method.
+Pending payments are also removed if the Wasabi client restarts.
+
+
+### listpaymentsincoinjoin
+
+Lists current payincoinjoin('s).
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:653]:
```suggestion
Lists the current payments in coinjoins.
```

File: docs/using-wasabi/RPC.md [hunk 5]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+
+```bash
+curl -s --data-binary '{"jsonrpc":"2.0", "id":"1", "method":"listpaymentsincoinjoin", "params":[]}' http://127.0.0.1:37128/WalletName | jq
+{
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:657]: Same as above, separate between the bash command and the json result.

File: docs/using-wasabi/RPC.md [hunk 6]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+      "address": "tb1qaznf0ky4yh8vc3yhew984jhxugr8apeu7wa98d"
+    }
+  ],
+  "id": "1"
+}
+```
+
+### cancelpaymentincoinjoin
+
+Cancel a _paymentincoinjoin_.
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:678]:
```suggestion
Cancels a payment in coinjoin.
```

File: docs/using-wasabi/RPC.md [hunk 7]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+
+```bash
+curl -s --data-binary '{"jsonrpc":"2.0", "id":"1", "method":"cancelpaymentincoinjoin", "params":["65c21ec1-9865-4cd6-bd67-c2f058a45d24"]}' http://127.0.0.1:37128/WalletName | jq
+{
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:682]: Same as above.

File: docs/using-wasabi/RPC.md [hunk 8]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+  "jsonrpc": "2.0"
+  "id": "1"
+}
+```
+
+The parameter is the paymentId of the pending payment to be cancelled.
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:688]:
```suggestion
The parameter is the paymentId of the pending payment in coinjoin to be cancelled.
```

File: docs/using-wasabi/RPC.md [hunk 9]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
...
+
```
Comment (yahiheb) [inline docs/using-wasabi/RPC.md:689]:
```suggestion
```

**Review by yahiheb — CHANGES_REQUESTED** [review]

File: docs/using-wasabi/RPC.md [hunk 10]
```diff
@@ -620,6 +620,79 @@ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"startcoinjoin", "para
 
 The first parameter is the wallet password, the second parameter is `stopWhenAllMixed`, and the third one is`overridePlebStop`.
 
+### payincoinjoin
+
+Pay to a bitcoin address in a coinjoin.
+
+```bash
+curl -s --data-binary '{"jsonrpc":"2.0", "id":"1", "method":"payincoinjoin", "params":["tb1qaznf0ky4yh8vc3yhew984jhxugr8apeu7wa98d", 10000]}' http://127.0.0.1:37128/WalletName | jq
+{
```
[the remark this answers is not in the document]
Comment (yahiheb) [reply docs/using-wasabi/RPC.md:644]: I meant to separate the command and the result completely, not just by a line.

**Review by yahiheb — APPROVED** [review]
tACK

File: docs/using-wasabi/RPC.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (yahiheb) [reply docs/using-wasabi/RPC.md:644]: https://github.com/zkSNACKs/WasabiDoc/pull/1680

**Pull request merged**